### PR TITLE
[minor] update HiveExternalCatalogVersionsSuite to test 2.4.0

### PR DESCRIPTION
**Pull request by cloud-fan**
## What changes were proposed in this pull request?

Since Spark 2.4.0 is released, we should test it in HiveExternalCatalogVersionsSuite

## How was this patch tested?

N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @gatorsmile @srowen

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala [hunk 1]
```diff
@@ -206,7 +206,7 @@ class HiveExternalCatalogVersionsSuite extends SparkSubmitTestUtils {
 
 object PROCESS_TABLES extends QueryTest with SQLTestUtils {
   // Tests the latest version of every release line.
-  val testingVersions = Seq("2.1.3", "2.2.2", "2.3.2")
+  val testingVersions = Seq("2.1.3", "2.2.2", "2.3.2", "2.4.0")
```
Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala]: when will we drop 2.1 officially?

Comment (srowen) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala]: 2.1 should be removed actually. At least according to our default 18-month support rule, it ended support in June 2018, and I see no desire for a 2.1.4 release. 2.2 will end in Jan 2019 according to the same guideline. It's not a hard rule, but probably the right guidance to follow.

Certainly, as this test takes a while and can be flaky I'd like to keep it to 3 versions.

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by dongjoon-hyun** [comment]
It's an irrelevant failure.
```
*** glibc detected *** /usr/java/jdk1.8.0_60/bin/java: free(): invalid pointer: 0x00000007c05d6828 ***
======= Backtrace: =========
/lib64/libc.so.6[0x347c475dee]
/lib64/libc.so.6[0x347c478c3d]
/usr/java/jdk1.8.0_60/jre/lib/amd64/server/libjvm.so(+0x62bda1)[0x7f287574dda1]
/usr/java/jdk1.8.0_60/jre/lib/amd64/server/libjvm.so(+0x9b1b5c)[0x7f2875ad3b5c]
/usr/java/jdk1.8.0_60/jre/lib/amd64/server/libjvm.so(+0x9b2856)[0x7f2875ad4856]
/usr/java/jdk1.8.0_60/jre/lib/amd64/server/libjvm.so(+0xac137f)[0x7f2875be337f]
/usr/java/jdk1.8.0_60/jre/lib/amd64/server/libjvm.so(+0xac1800)[0x7f2875be3800]
/usr/java/jdk1.8.0_60/jre/lib/amd64/server/libjvm.so(+0x91cb88)[0x7f2875a3eb88]
/lib64/libpthread.so.0[0x347cc07aa1]
/lib64/libc.so.6(clone+0x6d)[0x347c4e8bcd]
```

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by dongjoon-hyun** [comment]
Thank you, @cloud-fan and @srowen .

Merged to master.

**Pull request closed without merging**